packages/core/src/hooks/useLookupAddress.ts

```diff
@@ -17,22 +19,30 @@ import { useEthers } from './useEthers'
  *   <p>Account: {ens ?? account}</p>
  * )
  */
-export function useLookupAddress(address: string | undefined) {
+export function useLookupAddress(address: string | undefined, rave: boolean = false) {
   const { library } = useEthers()
   const [ens, setENS] = useState<string | null>()
   const [isLoading, setIsLoading] = useState(false)
   const [error, setError] = useState<Error | null>(null)
+  const rave = new Contract('0x14Ffd1Fa75491595c6FD22De8218738525892101', [{"inputs":[{"internalType":"address","name":"owner","type":"address"},{"internalType":"uint256","name":"index","type":"uint256"}],"name":"getName","outputs":[{"internalType":"string","name":"","type":"string"}],"stateMutability":"view","type":"function"}], provider);
 
   useEffect(() => {
     let mounted = true
 
     void (async () => {
       if (!library || !address) return
       try {
-        setIsLoading(true)
-        const resolved = await library.lookupAddress(address)
-        if (!mounted) return
-        setENS(resolved)
+	setIsLoading(true)
+	if (rave) {
```

Comment:
I don't think this will work as intended as `rave` variable here is of type `Contract`, not `boolean`. Please move declaration of the rave contract from line 27 inside this `if` clause.